File: python/two_sum_1.py
"""
Author: Resul Emre AYGAN
"""

from typing import List


class Solution:
    @staticmethod
    def two_sum(nums: List[int], target: int) -> List[int]:
        index_dict = {}

        for index, number in enumerate(nums):
            temp_number = target - number

            if temp_number in index_dict:
                return [index_dict[temp_number], index]

            index_dict[number] = index

File: go/two-sum-1.go
/*
Author: Resul Emre AYGAN
*/

package _go

func twoSum(nums []int, target int) []int {
	indexSet := make(map[int]int)

	for index, number := range nums {
		tempNumber := target - number

		if tempIndex, ok := indexSet[tempNumber]; ok {
			return []int{tempIndex, index}
		}

		indexSet[number] = index
	}
	return nil
}
